### PR TITLE
Implement new Logs Table using Apollo Client

### LINKED ISSUE #617: Point logs graphql query to the new logs schema

### PR DESCRIPTION
**Pull request by Kevin101Zhang**
substituted rest endpoints for logging tables using apollo client. 

https://www.loom.com/share/4dbea64625d2449dbd7b2c6e7e3eecc8?sid=cea1b56c-88a4-488a-8ab1-20cabc76ece4

**Known Issues:**

1. Eslint: React Hook useEffect has a missing dependency: 'renderGrid'. Either include it or remove the dependency array. This triggers the use of callbacks, which work fine for fetching/refetching data but cause conflicts with the Grid.js `search flag` functionality being responsive..

2. Occasionally, when you click between page, it may trigger a` 'IndexerLogs.jsx:113 [Grid.js] [ERROR]: The container element [object HTMLDivElement] is not empty. Make sure the container is empty and call render() again'.` This will not break the page, but numerous attempts to resolve the issue without direct DOM manipulation have been made. Due to overriding the form of pagination and adding our own, hidden conflicts that have not been resolved seem to exist.

I prefer we just move away from gridjs. Outside of its minimal table formatting we are doing (see gridConfig). The only real features we use are the search functionality. The library is not inherently bad - Its extremely lightweight but a poor choice in this case because we use apollo client as oppose to its more configurable counterpart requiring a REST endpoint. There is not much control we have when interacting with gridjs features. As a result - such features and homemade pagination cause issues due to potential conflicts in managing data rendering and state updates. 

**Odd behavior:**

I only seem to be getting a specific blockheight despite the timestamp and ID changing so I strongly believe im not fetching the same block. The IDs are unique but it lead me to think that it seems as we're repeatedly processing the same block again.

This is what the page looks like if query failed. The console also shows the error message. The query can mainly fail if either the indexer has not run once since we first added in provisioning for the new tables (thus the tables don't exist), or the provisioning failed.  
image

### CODE REVIEW

**Comment by Kevin101Zhang** [comment]
Links to look into:
https://github.com/grid-js/gridjs/issues/44
https://stackoverflow.com/questions/76323920/custom-pagination-in-gridjs-with-react

**Review by darunrs — COMMENTED** [review]
I will try out the changes locally and see. I'll commit changes for my comments and merge the PR, since Kevin is off and we need the changes into dev ASAP.

File: frontend/src/components/Logs/IndexerLogs.jsx [hunk 1]
```diff
@@ -1,241 +1,187 @@
...
+  const functionName = `${indexerDetails.accountId}/${indexerDetails.indexerName}`;
+  const hasuraAccountId = indexerDetails.accountId.replace(/\./g, "_");
+  const schemaName = `${hasuraAccountId}_${indexerDetails.indexerName}`;
+  const tableName = `${schemaName}_sys_logs`;
+
+  const GET_INDEXER_LOGS = gql`
+    query GetIndexerLogs($limit: Int, $offset: Int) {
+      ${tableName}(limit: $limit, offset: $offset, order_by: {timestamp: desc}) {
+        block_height
+        date
```
[the remark this answers is not in the document]
Comment (darunrs) [reply frontend/src/components/Logs/IndexerLogs.jsx:24]: For now I'll leave this be.

**Comment by darunrs** [comment]
Couple general problems that hopefully we can resolve in the near future. 

It seems like the search bar only applies to the currently loaded page? And not all logs. This obviously renders the search bar largely useless. 

We lost the nice visual touch of hiding the row separation for the logs that share block heights. Since block height is no longer the primary index, we'll have to rethink the design. For now, it works though. 

As mentioned, we probably don't need the Date in addition to timestamp? It would be nice if we could improve the way we show the timestamp though as it's hard to read as is.

**Pull request merged**